runner/cf-executor/cg/cg_test.go

```diff
@@ -0,0 +1,293 @@
+package cg
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+type adapterStub struct {
+	CloudI
+
+	StCreds *Creds
+	StURL   string
+	StApps  []*App
+
+	FailGetApps bool
+	FailConnect bool
+}
+
+func (a *adapterStub) getApps() (apps []*App, err error) {
+	if a.FailGetApps {
+		return nil, errors.New("fail")
+	}
+	return a.StApps, nil
+}
+
+func (a *adapterStub) connect(url string, creds *Creds) (_ error) {
+	if a.FailConnect {
+		return errors.New("fail")
+	}
+	a.StURL = url
+	a.StCreds = creds
+	return nil
+}
+
+type credIStub struct {
+	U    string
+	P    string
+	Fail bool
+}
+
+func (c credIStub) getCreds() (*Creds, error) {
+	if c.Fail {
+		return nil, errors.New("fail")
+	}
+	return &Creds{c.U, c.P}, nil
+}
+
+func TestNew(t *testing.T) {
+	optsStub := &Opts{CredI: credIStub{"a", "b", false}}
+	cgStub := &CG{&adapterStub{
+		StURL:   apiRootURLDefault,
+		StCreds: &Creds{"a", "b"},
+	}, optsStub}
+
+	tests := []struct {
+		want    *CG
+		opts    *Opts
+		wantErr interface{}
+		name    string
+	}{
+		{name: "fails using default credential getter", wantErr: &syntaxError},
```

Comment:
[issue (non-blocking)] this, and some of the creds tests, are fairly brittle in that they'll fail if run with `CF_USERNAME` and/or `CF_PASSWORD` set. I think this is something we'll want to return to if not addressed here

Reply:
For some reason I had been thinking that the test environment was isolated. It is not! Will fix this before merging, especially because I think it wouldn't be unusual for these to be defined for some other reason in CI.